corehq/apps/reports/formdetails/readable.py

```diff
@@ -761,8 +764,7 @@ def _compile_form(self, form):
         except XFormException as exception:
             form_meta['questions'] = []
             form_meta['error'] = {
-                'details': six.text_type(exception),
-                'edit_url': reverse('form_source', args=[self.domain, self.app._id, form.unique_id]),
+                'details': six.text_type(exception.args[0]),
```

Comment:
Why doesn't `six.text_type(exception)` work here?

Reply:
The place where this exception is raised gives the exception two arguments: https://github.com/dimagi/commcare-hq/blob/76eebaa86b78e557783b3cab8e82b6263d9266b4/corehq/apps/app_manager/models.py#L1147. If you `six.text_type` on that it just prints a tuple of both args.
```py
In [4]: try:
   ...:     raise Exception("hello", "blah")
   ...: except Exception as e:
   ...:     print six.text_type(e)
   ...:     
   ...:     
('hello', 'blah')
```
I agree that it is pretty gross... Do you think it's worth refactoring that exception? I didn't look too closely at where that is caught or what the implications of changing how it is raised would be.

Reply:
Weird. It looks like the other places that raise `XFormException` just give it a string arg, and the places that catch it usually treat it like it's just a string, not a tuple (I didn't check every instance). Having an exception raised in two different ways seems...cruel.

I'd guess it's safe to change the raise to be like this: https://github.com/dimagi/commcare-hq/blob/0b2372ba5dc1cb1638a1b4eede06fed4a6cae575/corehq/apps/app_manager/models.py#L5223

...but I also think it'd be good to double-check all of the places that raise or catch this exception, and I understand if that's out of scope of this PR.